### PR TITLE
Fix: StaggeredLayout division bug

### LINKED ISSUE #542: StaggeredLayout crashes on ARM64 build
### Describe the bug

The `StaggeredLayout` from the new namespace `CommunityToolkit.WinUI.Controls.Primitives` crashes on **ARM64** build when used in an ItemsView with the following StackTace:

```
Message = "Array dimensions exceeded supported range."

   at CommunityToolkit.WinUI.Controls.StaggeredLayout.MeasureOverride(VirtualizingLayoutContext context, Size availableSize)
   at Microsoft.UI.Xaml.Controls.VirtualizingLayout.Microsoft.UI.Xaml.Controls.IVirtualizingLayoutOverrides.MeasureOverride(VirtualizingLayoutContext context, Size availableSize)
   at ABI.Microsoft.UI.Xaml.Controls.IVirtualizingLayoutOverrides.Do_Abi_MeasureOverride_2(IntPtr thisPtr, IntPtr context, Size availableSize, Size* result)
```

Note that the sample app also crashes if you go to the StaggeredLayout page (IF you are on **ARM64** device).

### Regression

_No response_

### Reproducible in sample app?

- [x] This bug can be reproduced in the sample app.

### Steps to reproduce

```text
1. Build and run the sample app on ARM64.
2. Observe the crash.

Here's the link for the sample app: https://github.com/euju-ms/StaggeredLayoutArm64

The code basically uses the StaggeredLayout in ItemsView.

<ItemsView ...>

    <ItemsView.Layout>
        <tkControls:StaggeredLayout />
    </ItemsView.Layout>
</ItemsView>
```

### Expected behavior

Expected to see three items show up instead of crashing.

### Screenshots

Following three blue squares should show when it works; but using the newest StaggeredLayout crashes the app immediately.

image

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [X] Other (specify)

### Other Windows Build number

Windows 11 24H2 (Build 26100.1742)

### App minimum and target SDK version

- [X] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

2022

### Visual Studio Build Number

Version 17.11.3

### Device form factor

Desktop

### Nuget packages

```
<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.6.240829007" />
<PackageReference Include="CommunityToolkit.WinUI.Controls.Primitives" Version="8.1.240916" />
```

### Additional context

I set `WindowsSdkPackageVersion` of `10.0.22621.45` for WinAppSdk 1.6 to work (but I verified that the issue persists without this in  WinAppSdk 1.5.240802000).

I have also verified that the StaggeredLayout from the old package `CommunityToolkit.WinUI.UI.Controls` version `7.1.2` works.

### Help us help you

No.

### PR DESCRIPTION
**Pull request by Poker-sang**
## Fixes

closes #542 
fix <https://github.com/microsoft/PowerToys/pull/35139>

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements: 

- [x] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [x] Based off latest main branch of toolkit
- [x] Tested code with current supported SDKs
- [ ] New component
  - [ ] Documentation has been added
  - [ ] Sample in sample app has been added
  - [ ] Analyzers are passing for documentation and samples
  - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/CommunityToolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (if applicable)
- [ ] Header has been added to all new source files
- [x] Contains **NO** breaking changes

### CODE REVIEW

**Comment by michael-hawker** [comment]
Not sure what happened with the tooling submodule here either.

Wondering if we could add a test for anything around this to help guard future issues?

**Review by michael-hawker — APPROVED** [review]
🦙❤️ Thanks!

File: components/Primitives/src/StaggeredLayout/StaggeredLayout.cs [hunk 1]
```diff
@@ -163,7 +163,7 @@ protected override Size MeasureOverride(VirtualizingLayoutContext context, Size
         if (ItemsStretch is StaggeredLayoutItemsStretch.None)
         {
             columnWidth = double.IsNaN(DesiredColumnWidth) ? availableWidth : Math.Min(DesiredColumnWidth, availableWidth);
-            numColumns = Math.Max(1, (int)Math.Floor(availableWidth / state.ColumnWidth));
+            numColumns = Math.Max(1, (int)Math.Floor(availableWidth / (DesiredColumnWidth + ColumnSpacing)));
```
Comment (michael-hawker) [inline components/Primitives/src/StaggeredLayout/StaggeredLayout.cs:166]: Oh @Poker-sang just noticed, originally this was just `columnWidth` from above, should we still be using that as it does the NaN check? 

We'd just move the `ColumnSpacing` as part of that value on 165? I guess that was missing before? (Maybe it'd be better to move that add of the ColumnSpacing change to #573, and just keep this change to be the fix for the change from `state.`?) Then it's easy to approve the variable substitution here quick for the crash, and we can just re-validate the other behavior changes all within #573 together?

**Review by michael-hawker — CHANGES_REQUESTED** [review]
Was just about to approve and merge, but noticed it was slightly different than when I had first looked before the weekend. Thanks for fixing the tooling submodule stuff.

**Comment by Poker-sang** [comment]
Ok reverted it. i commit this before because i found that we forget to calculate spacing before. And we might see the bug clearly when `ColumnSpacing` is close to `DesiredColumnWidth`. And you are right we should make it to #573

**Review by michael-hawker — APPROVED** [review]

**Comment by michael-hawker** [comment]
🎉 Thanks @Poker-sang for this quick fix! 🦙❤️ And for moving the column fix to the other PR. I'll get this merged once the CI passes, some strange issue I haven't seen before there; so just re-kicking to see if it was just a stray cosmic ray! 🤞

**Pull request merged**